db/db_creator.py

```diff
@@ -41,34 +42,62 @@ def __init__(
         )
         self.engine: Engine = create_engine(self.conn_str)
 
-        self.append_table_names: List[str] = DEFAULT_APPEND_TABLE_NAMES
-        if append_table_names is not None:
-            self.append_table_names += append_table_names
-
     def get_table_names(self) -> List[str]:
+        '''
+        Gets table names using the SQLAlchemy Engine object.
+        '''
         logger.debug('get_table_names')
         return self.engine.table_names()
 
     def migrate(self) -> DBCreator:
+        '''
+        Updates database schema using yoyo-migrations and the migration files in the
+        migrations directory collocated with this file (db_creator.py).
+        '''
         logger.debug('migrate')
         backend = get_backend(self.conn_str)
         migrations = read_migrations(MIGRATIONS_PATH)
         with backend.lock():
             backend.apply_migrations(backend.to_apply(migrations))
         return self
 
-    def drop_records(self) -> DBCreator:
+    def drop_records(self, spec_table_names: Union[Sequence[str], None] = None) -> DBCreator:
+        '''
+        Drops records from either the specified database tables or all of the
+        application-managed database tables; tables managed by yoyo-migrations are ignored.
+        '''
         logger.debug('drop_records')
+        app_table_names = [
+            table_name for table_name in self.get_table_names() if 'yoyo' not in table_name
+        ]
+        logger.debug(f'app_table_names: {app_table_names}')
+
+        if spec_table_names is None:
+            # Drop all non-yoyo tables
+            logger.info('Dropping all application (non-yoyo) tables')
+            drop_table_names = app_table_names
+        else:
+            # Drop specified table names if they're valid (i.e. an application-managed table)
+            logger.info('Dropping specific provided tables if they are valid')
+            drop_table_names = []
+            for spec_table_name in spec_table_names:
+                if spec_table_name in app_table_names:
+                    drop_table_names.append(spec_table_name)
+                else:
+                    logger.error(f'Invalid table name was provided: {spec_table_name}')
+
         conn = self.engine.connect()
         conn.execute('SET FOREIGN_KEY_CHECKS=0;')
-        for table_name in self.get_table_names():
-            if 'yoyo' not in table_name and table_name not in self.append_table_names:
-                logger.debug(f'Table Name: {table_name}')
-                conn.execute(f'DELETE FROM {table_name};')
-                logger.info(f'Dropped records in {table_name} in {self.db_name}')
+        for drop_table_name in drop_table_names:
+            logger.debug(f'Table Name: {drop_table_name}')
+            conn.execute(f'DELETE FROM {drop_table_name};')
+            logger.info(f'Dropped records in {drop_table_name} in {self.db_name}')
         conn.execute('SET FOREIGN_KEY_CHECKS=1;')
```

Comment:
@ssciolla if the DB traction failed for some here we want it to be handled. Otherwise, Foreign constraints will be effected. 

In general, I have seen that whenever we are writing data using pandas SQL Alchemy we are not surrounding with try-catch. we have to review all the cases with DB transaction and see is proper handling is implemented if not here but open an issue and deal with it later.

Reply:
Okay, so what do you mean by handled? Just to catch the exception and report it? How would that be different from just looking for an exception? I understand the possible concern of tables getting messed up, but that would happen regardless I think. MySQL doesn't support rollbacks of transactions. I'm just trying to understand what wrapping things in `try/except` accomplishes.

Also, do you see this is a a significant issue? If so, we should be addressing it before we release. I don't have a good sense of how big of a problem you see this as. We haven't talked about it before, even though we've worked on code together that uses `to_sql`.

Reply:
I am seeing that MyLA does wrap things in `try/except` in `cron.py`, so I'm looking there for some explanation/clarity.

Reply:
If one query failed, I don't think we'd want the others to continue, right? It's all interconnected.

Reply:
Yeah.. that's the idea. If we log it as an error with try-catch then Jenkin sees that as error and reports it. At the minimum we might need to do is rerun the cron to fix it.

Reply:
I still don't see how that would be different from just letting it error out; Jenkins would still log it as a failure.

I am seeing that there might be ways to support rollbacks with MySQL and Python, but I'm not sure yet if there's a way to do it with `pandas`.

Would you mind creating an issue for this? I think we can have this discussion there, and if we need to come up with an emergency fix, I can work on that.